### PR TITLE
Feat: collect app status from all envs

### PR DESCRIPTION
**Pull request by wangyikewxgm**
### Description of your changes

collect app status from all envs

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [ ] Run `yarn lint` to ensure the frontend changes are ready for review.
- [x] Run `make reviewable`to ensure the server changes are ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/server/domain/service/application.go [hunk 1]
```diff
@@ -315,6 +316,35 @@ func (c *applicationServiceImpl) GetApplicationStatus(ctx context.Context, appmo
 	return &app.Status, nil
 }
 
+// GetApplicationStatusFromAllEnvs get applications status from all envs
+func (c *applicationServiceImpl) GetApplicationStatusFromAllEnvs(ctx context.Context, appmodel *model.Application) ([]*apisv1.ApplicationStatusResponse, error) {
+	envBindings, err := c.EnvBindingService.GetEnvBindings(ctx, appmodel)
```
Comment (barnettZQG) [inline pkg/server/domain/service/application.go:321]: Can we reuse the GetApplicationCR function?

File: pkg/server/domain/service/application.go [hunk 2]
```diff
@@ -1779,6 +1809,10 @@ func (c *applicationServiceImpl) RollbackWithRevision(ctx context.Context, appli
 	}, nil
 }
 
+func (c *envBindingServiceImpl) GetApplicationStatus() {
```
Comment (barnettZQG) [inline pkg/server/domain/service/application.go:1812]: Please remove the empty function.

File: pkg/server/interfaces/api/application.go [hunk 3]
```diff
@@ -433,6 +433,16 @@ func (c *application) GetWebServiceRoute() *restful.WebService {
 		Returns(400, "Bad Request", bcode.Bcode{}).
 		Writes(apis.ApplicationStatusResponse{}))
 
+	ws.Route(ws.GET("/{appName}/status").To(c.getApplicationStatusFromAllEnvs).
+		Doc("get application status from all envs").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Filter(c.RbacService.CheckPerm("envBinding", "detail")).
+		Filter(c.appCheckFilter).
+		Param(ws.PathParameter("appName", "identifier of the application ").DataType("string")).
+		Returns(200, "OK", []*apis.ApplicationStatusResponse{}).
```
Comment (barnettZQG) [inline pkg/server/interfaces/api/application.go:442]: The response data is imperfection. Can't distinguish between different environments.

Comment (wangyikewxgm) [reply pkg/server/interfaces/api/application.go:442]: This struct is as follows, it contain the the filed of  envName:

```json
type ApplicationStatusResponse struct {
	EnvName string            `json:"envName"`
	Status  *common.AppStatus `json:"status"`
}
```

File: pkg/server/domain/service/application.go (hunk 1 above)
Comment (wangyikewxgm) [reply pkg/server/domain/service/application.go:321]: GetApplicationCR only return all applications that will loss env info.

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**